scheduler/reconcile.go

```diff
@@ -731,16 +740,25 @@ func (a *allocReconciler) computePlacements(group *structs.TaskGroup,
 // The input deploymentPlaceReady is calculated as the deployment is not paused, failed, or canarying.
 // It returns the number of allocs still needed.
 func (a *allocReconciler) computeReplacements(deploymentPlaceReady bool, desiredChanges *structs.DesiredUpdates,
-	place []allocPlaceResult, failed, lost allocSet, underProvisionedBy int) int {
+	place []allocPlaceResult, rescheduleNow, lost allocSet, underProvisionedBy int) int {
+
+	stop := make(allocSet, 0)
+	// Disconnecting allocs are not failing.
+	for id, alloc := range rescheduleNow {
```

Comment:
This looks wrong. I'm taking a look at it now, but don't want to hold up the PR review.

Reply:
Adding "reschedule now" to the stop set seems to make sense. We need to server-stop allocs that we're replacing (which might only be client-failed), right?

Reply:
I remember why this is here now, but I didn't comment on it well originally. I reviewed the code and am adding the following comments to clarify. I'm also renaming `stop` to `failed` so that it clarifies the intent of this function.

```go
// Disconnecting allocs are not failing, but are included in rescheduleNow.
// Create a new set that only includes the actual failures and compute
// replacements based off that.
```